src/org/olap4j/driver/xmla/XmlaOlap4jCube.java

```diff
@@ -47,8 +47,9 @@
         new HashMap<String, XmlaOlap4jHierarchy>();
     final Map<String, XmlaOlap4jLevel> levelsByUname =
         new HashMap<String, XmlaOlap4jLevel>();
-    final List<XmlaOlap4jMeasure> measures =
-        new ArrayList<XmlaOlap4jMeasure>();
+    final NamedList<XmlaOlap4jMeasure> measures;
+    final Map<String, XmlaOlap4jMeasure> measuresMap =
```

Comment:
Team, I wanted to know if someone have done or is willing to do unit testing for this change in the code. as It is more often I hear cases related to the First XMLA query to take long time.